concordia/tasks.py

```diff
@@ -89,9 +90,84 @@ def site_report():
     for campaign in Campaign.objects.all():
         campaign_report(campaign)
 
+    for topic in Topic.objects.all():
+        topic_report(topic)
 
-def campaign_report(campaign):
 
+def topic_report(topic):
+    report = {
+        "assets_not_started": 0,
+        "assets_in_progress": 0,
+        "assets_submitted": 0,
+        "assets_completed": 0,
+    }
+
+    asset_count_qs = (
+        Asset.objects.filter(item__project__topics=topic)
+        .values_list("transcription_status")
+        .annotate(Count("transcription_status"))
+    )
+
+    for status, count in asset_count_qs:
+        logger.debug("Topic %s assets %s: %d", topic.slug, status, count)
+        report[f"assets_{status}"] = count
+
+    assets_total = Asset.objects.filter(item__project__topics=topic).count()
+    assets_published = (
+        Asset.objects.published().filter(item__project__topics=topic).count()
+    )
+    assets_unpublished = (
+        Asset.objects.unpublished().filter(item__project__topics=topic).count()
+    )
+
+    items_published = Item.objects.published().filter(project__topics=topic).count()
+    items_unpublished = Item.objects.published().filter(project__topics=topic).count()
+
+    projects_published = Project.objects.published().filter(topics=topic).count()
+    projects_unpublished = Project.objects.unpublished().filter(topics=topic).count()
+
+    anonymous_transcriptions = Transcription.objects.filter(
+        asset__item__project__topics=topic, user=get_anonymous_user()
+    ).count()
+    transcriptions_saved = Transcription.objects.filter(
+        asset__item__project__topics=topic
+    ).count()
+
+    asset_tag_collections = UserAssetTagCollection.objects.filter(
+        asset__item__project__topics=topic
+    )
+
+    stats = asset_tag_collections.order_by().aggregate(tag_count=Count("tags"))
+    tag_count = stats["tag_count"]
+
+    distinct_tag_list = set()
+
+    for tag_collection in asset_tag_collections:
+        distinct_tag_list.add(tag_collection.tags.values_list("pk", flat=True))
```

Comment:
If I'm reading this correctly, it should be calling `set.update` to add each `pk` value to the set rather than each unique group of pk values:

```suggestion
        distinct_tag_list.update(tag_collection.tags.values_list("pk", flat=True))
```

If I'm not rushing this review too much, we'd also want to update https://github.com/LibraryOfCongress/concordia/blob/de024d80f015a65cbdc3e66eb465968430a350a1/concordia/tasks.py#L148

Reply:
I think they both do the same thing, but `.update` is clearer.